app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt

```diff
@@ -0,0 +1,147 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.profile.ProfileChooserFragment
+import org.robolectric.annotation.Config
+
+private const val TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLERVIEW = "profile_recycler_view"
+
+/**
+ * Tests for ensuring [ProfileChooserFragment] uses the correct column count for profiles based on display density.
+ * document reffered :https://developer.android.com/reference/androidx/test/core/app/ActivityScenario
+ * */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class ProfileChooserSpanTest {
+
+  private lateinit var context: Context
+  private var activity: ProfileChooserFragmentTestActivity? = null
+  private lateinit var fragment: ProfileChooserFragment
+  private lateinit var recyclerView: RecyclerView
+
+  @Before
+  @ExperimentalCoroutinesApi
+  fun setUp() {
+    Intents.init()
+    activity = getProfileChooserTestActivity()
+    fragment = getProfileChooserFragment()
+    recyclerView = (fragment!!.view!!.findViewWithTag<View>(TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLERVIEW) as RecyclerView)
+    ApplicationProvider.getApplicationContext<Context>().resources.configuration.orientation =
+      Configuration.ORIENTATION_LANDSCAPE
+    context = ApplicationProvider.getApplicationContext()
+  }
+
+  @get:Rule
+  var activityRule = ActivityScenarioRule(ProfileChooserFragmentTestActivity::class.java)
+
+  private fun getProfileChooserTestActivity(): ProfileChooserFragmentTestActivity? {
+    activityRule.scenario.onActivity {
+      activity = it
+    }
+    return activity as ProfileChooserFragmentTestActivity
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Test
+  @Config(qualifiers = "ldpi")
+  fun testProfileChooserSpanTest_onConfigLandScapeAndScreenldpi() {
+    assertThat(context.resources.getInteger(R.integer.profile_chooser_span_count)).isEqualTo(3)
+  }
+
+  @Test
+  @Config(qualifiers = "mdpi")
+  fun testProfileChooserSpanTest_onConfigLandScapeAndScreenMdpi() {
+    assertThat(context.resources.getInteger(R.integer.profile_chooser_span_count)).isEqualTo(3)
+  }
+
+  @Test
+  @Config(qualifiers = "hdpi")
+  fun testProfileChooserSpanTest_onConfigLandScapeAndScreenHdpi() {
+    assertThat(context.resources.getInteger(R.integer.profile_chooser_span_count)).isEqualTo(4)
+  }
+
+  @Test
+  @Config(qualifiers = "xhdpi")
+  fun testProfileChooserSpanTest_onConfigLandScapeAndScreenXhdpi() {
+    assertThat(context.resources.getInteger(R.integer.profile_chooser_span_count)).isEqualTo(5)
+  }
+
+  @Test
+  @Config(qualifiers = "xxxhdpi")
+  fun testProfileChooserSpanTest_onConfigLandScapeAndScreenXxxhdpi() {
+    assertThat(context.resources.getInteger(R.integer.profile_chooser_span_count)).isEqualTo(5)
+  }
+
+  @Test
+  @Config(qualifiers = "land-ldpi")
+  fun ProfileChooserFragmentTest_onConfigLandScapeAndForLdpi_hasRecyclerViewSpanCountVerifiedSucessfully() {
```

Comment:
Some parts of this test name could be improved, e.g.:

```kotlin
testProfileChooserFragmentRecyclerView_landscape_ldpi_hasCorrectSpanCount
```

Please apply this test pattern for each of the following tests.

Reply:
Done.